### PR TITLE
Allow editing of overwrites in Channel.edit()

### LINKED ISSUE #2197: Allow editing overwrites through channel.edit()
### The Problem

This will allow to update all overwrites of a channel at the same time.

### The Ideal Solution

`overwrites` kwarg in `channel.edit()` which will work the same as the one used in `Guild.create_text_channel()`
Discord API supports updating all overwrites of a channel at the same time using the very same endpoint: https://discordapp.com/developers/docs/resources/channel#modify-channel

### The Current Solution

Using `channel.set_permissions()` for each overwrites target separately, which requires more http requests.

### Summary

Basically this:
```py
overwrites = {
    guild.default_role: discord.PermissionOverwrite(read_messages=False),
    guild.me: discord.PermissionOverwrite(read_messages=True)
}

await channel.edit(overwrites=overwrites)
```

### PR DESCRIPTION
**Pull request by CapnS**
### Summary

Allows the editing of permission overwrites in Channel.edit(). Resolves #2197 

### Checklist

- [X] If code changes were made then they have been tested.
    - [X] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [X] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/abc.py [hunk 1]
```diff
@@ -256,6 +256,28 @@ def _sorting_bucket(self):
                 options['permission_overwrites'] = [c._asdict() for c in category._overwrites]
         else:
             await self._move(position, parent_id=parent_id, lock_permissions=lock_permissions, reason=reason)
+        
+        overwrites = options.pop('overwrites')
```
Comment (Rapptz) [inline discord/abc.py]: This raises `KeyError` if not given.

**Comment by CapnS** [comment]
I went through and made the changes requested.

File: discord/channel.py [hunk 2]
```diff
@@ -867,11 +876,26 @@ def is_nsfw(self):
             category.
         reason: Optional[:class:`str`]
             The reason for editing this channel. Shows up on the audit log.
-
+        overwrites: :class:`dict`
+            A :class:`dict` of target (either a role or a member) to
+            :class:`PermissionOverwrite` to apply to the channel.    
+            .. versionadded:: 1.3
```
Comment (Rapptz) [inline discord/channel.py]: Needs a newline.

File: discord/channel.py [hunk 3]
```diff
@@ -867,11 +876,26 @@ def is_nsfw(self):
...
+       
+        Example
```
Comment (Rapptz) [inline discord/channel.py]: Probably should remove this example. It's too destructive to have such a simple example giving people the wrong expectation.

**Comment by CapnS** [comment]
Resolved the new changes requested

**Pull request closed without merging**